madsim-rdkafka/src/std/client.rs

```diff
@@ -354,29 +354,32 @@ impl<C: ClientContext> Client<C> {
     }
 
     /// Returns high and low watermark for the specified topic and partition.
-    pub async fn fetch_watermarks<T: Into<Timeout>>(
+    pub async fn fetch_watermarks<T: Into<Timeout> + Send + 'static>(
         &self,
         topic: &str,
         partition: i32,
         timeout: T,
     ) -> KafkaResult<(i64, i64)> {
-        let mut low = -1;
-        let mut high = -1;
         let topic_c = CString::new(topic.to_string())?;
-        let ret = unsafe {
-            rdsys::rd_kafka_query_watermark_offsets(
-                self.native_ptr(),
+        let native_client = unsafe { NativeClient::from_ptr(self.native_ptr()) };
```

Comment:
`NativeClient` will call `rd_kafka_destroy` when it's get dropped (behavior of `NativePtr`). Will this lead to problem?

Reply:
Ohhh. I think so. Thanks for your troubleshooting!